sim/team-validator.ts

```diff
@@ -749,8 +749,8 @@ export class TeamValidator {
 		for (const moveName of set.moves) {
 			const move = dex.getMove(moveName);
 			if (move.id === 'hiddenpower' && move.type !== 'Normal') {
-				if (!set.hpType) {
-					set.hpType = move.type;
+				if (!set.hpType || this.toID(set.hpType) === 'undefined') {
```

Comment:
I believe this is entirely unnecessary. The `sim/dex` changes are all that are needed; this just slows down validation.

Reply:
i blame sleep deprivation